lib/versioning/pep440/range.js

```diff
@@ -1,11 +1,113 @@
+
+const {
+  gte,
+  lte,
+  parse,
+  satisfies,
+} = require('@renovate/pep440');
+
+const {
+  parse: parseRange,
+} = require('@renovate/pep440/lib/specifier');
+
+
 module.exports = {
   getNewValue,
 };
 
+function getFutureVersion(baseVersion, toVersion) {
+  const toRelease = parse(toVersion).release;
+  const baseRelease = parse(baseVersion).release;
+  let found = false;
+  const futureRelease = baseRelease.map((basePart, index) => {
+    if (found) {
+      return 0;
+    }
+    const toPart = toRelease[index] || 0;
+    if (toPart > basePart) {
+      found = true;
+      return toPart + 1;
+    }
+    return toPart;
+  });
+  if (!found) {
+    futureRelease[futureRelease.length - 1] += 1;
+  }
+  return futureRelease.join('.');
+}
+
 function getNewValue(currentValue, rangeStrategy, fromVersion, toVersion) {
-  if (rangeStrategy === 'pin' || currentValue.startsWith('==')) {
+  // easy pin
+  if (rangeStrategy === 'pin') {
     return '==' + toVersion;
   }
-  logger.warn('Unsupported currentValue: ' + currentValue);
-  return toVersion;
+  const ranges = parseRange(currentValue);
+  if (!ranges) {
+    logger.warn('Invalid currentValue: ' + currentValue);
+    return null;
+  }
+  if (!ranges.length) {
+    // an empty string is an allowed value for PEP440 range
+    // it means get any version
+    logger.warn('Empty currentValue: ' + currentValue);
+    return currentValue;
+  }
+  if (rangeStrategy === 'replace') {
+    if (satisfies(toVersion, currentValue)) {
```

Comment:
I can’t think of a case where this should happen but it’s good to leave in the logic just in case.